### PR TITLE
MAINT: Remove Malware Domain List feed

### LINKED ISSUE #1762: Malware Domain List feed broken
As reported on the mailing list, the feed Malware Domain List is broken, it returns an empty page.
https://lists.cert.at/pipermail/intelmq-users/2021-February/000202.html ff.

It is not clear if this is a temporary issue or permanent.

If it's temporary, the parser should be able to handle that, otherwise we can remove it.

### PR DESCRIPTION
**Pull request by schacht-certat**
This commit removes the malwaredomainlist parser bot and all relevant
unit tests. It removes the relevant feed entry in the feeds.yaml file
and drops the malwaredomainlist bot entry in the BOTS file and in the
runtime.conf.
Last but not least it adds two methods to the upgrades.py removing bots
that use the malwaredomainlist parser module and pointing out collectors
that have the malwaredomain list url set.

Closes #1762

### CODE REVIEW

File: intelmq/lib/upgrades.py [hunk 1]
```diff
@@ -532,6 +534,41 @@ def v230_csv_parser_parameter_fix_1(defaults, runtime, harmonization, dry_run):
...
+    """
+    Deprecate malwaredomainlist parser
+    """
+    changed = None
+    malwaredomainlistparser = []
+    for bot_id, bot in runtime.items():
+        if bot["module"] == "intelmq.bots.parsers.malwaredomainlist.parser":
+            malwaredomainlistparser.append(bot_id)
+    for bot_id in malwaredomainlistparser:
+        del runtime[bot_id]
```
Comment (wagner-certat) [inline intelmq/lib/upgrades.py:547]: That could cause troubles as the pipeline and (to a lesser degree) the positions need to be adapted as well.

File: intelmq/lib/upgrades.py [hunk 2]
```diff
@@ -550,7 +587,7 @@ def v230_csv_parser_parameter_fix_1(defaults, runtime, harmonization, dry_run):
     ((2, 2, 2), (v222_feed_changes, )),
     ((2, 2, 3), ()),
     ((2, 3, 0), (v230_csv_parser_parameter_fix_1, )),
-    ((3, 0, 0), ()),
+    ((3, 0, 0), (v300_feed_changes, v300_deprecations,)),
```
Comment (wagner-certat) [inline intelmq/lib/upgrades.py:590]: Can you please move the functions to 2.3.0? They are a bug fix IMHO.

**Review by wagner-certat — CHANGES_REQUESTED** [review]
Please also remove the bots from `intelmq/etc/pipeline.conf`.

Could you please add a test case for the upgrade function?

File: intelmq/lib/upgrades.py (hunk 1 above)
Comment (schacht-certat) [reply intelmq/lib/upgrades.py]: Fixed, it doesn't change anything, but emits a warning

File: intelmq/lib/upgrades.py (hunk 2 above)
Comment (schacht-certat) [reply intelmq/lib/upgrades.py]: done

**Comment by schacht-certat** [comment]
> Please also remove the bots from `intelmq/etc/pipeline.conf`.
> 
> Could you please add a test case for the upgrade function?

done and done

**Review by wagner-certat — APPROVED** [review]

**Pull request merged**